keyboards/mxss/rules.mk

```diff
@@ -0,0 +1,75 @@
+# MCU name
+#MCU = at90usb1286
+MCU = atmega32u4
+
+# Processor frequency.
+#     This will define a symbol, F_CPU, in all source code files equal to the
+#     processor frequency in Hz. You can then use this symbol in your source code to
+#     calculate timings. Do NOT tack on a 'UL' at the end, this will be done
+#     automatically to create a 32-bit value in your source code.
+#
+#     This will be an integer division of F_USB below, as it is sourced by
+#     F_USB after it has run through any CPU prescalers. Note that this value
+#     does not *change* the processor frequency - it should merely be updated to
+#     reflect the processor speed set externally so that the code can use accurate
+#     software delays.
+F_CPU = 16000000
+
+
+#
+# LUFA specific
+#
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+#     This will define a symbol, F_USB, in all source code files equal to the
+#     input clock frequency (before any prescaling is performed) in Hz. This value may
+#     differ from F_CPU if prescaling is used on the latter, and is required as the
+#     raw input clock is fed directly to the PLL sections of the AVR for high speed
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```

Comment:
Is this a Pro Micro, Teensy, or SMB MCU? 

aka, do you know which bootloader it's using?

Reply:
Bootloader is the Atmel DFU bootloader.

Reply:
Okay, then no issue with leaving this as is. 

However, if you want (optional), you can replace this line with `BOOTLOADER = atmel-dfu`, as the opt_defs line will be set later:
https://github.com/qmk/qmk_firmware/blob/6bd2b8ded3d1324bfb3e29e7e48f440835e466cd/bootloader.mk#L32-L41

As I said, this it optional.

Reply:
I feel that enforcing this in the template would be a good start if you want it to be used, I found bootloader.mk and mostly understood what it was doing, but left my rules.mk as it is to prevent any complications from unexpected behaviour. Things like this need to be documented if they're expected to be used.

Reply:
It is documented, but yeah, not well. 

Which is why I make sure that I flag and mention it. But for DFU bootloaders, there is no issue, as the "default" works.  It's Caterina, and halfkay  (Pro Micros and Teensy's, respectively) that have an issue with the default method and require special handling.